shell/common/shell.cc

```diff
@@ -457,18 +457,21 @@ void Shell::OnPlatformViewCreated(std::unique_ptr<Surface> surface) {
   // This is a synchronous operation because certain platforms depend on
   // setup/suspension of all activities that may be interacting with the GPU in
   // a synchronous fashion.
-
   fml::AutoResetWaitableEvent latch;
-  auto gpu_task = fml::MakeCopyable([rasterizer = rasterizer_->GetWeakPtr(),  //
-                                     surface = std::move(surface),            //
-                                     &latch]() mutable {
-    if (rasterizer) {
-      rasterizer->Setup(std::move(surface));
-    }
-    // Step 3: All done. Signal the latch that the platform thread is waiting
-    // on.
-    latch.Signal();
-  });
+  auto gpu_task =
+      fml::MakeCopyable([this, rasterizer = rasterizer_->GetWeakPtr(),  //
```

Comment:
Is it possible to just capture `waiting_for_first_frame_` by reference but not the whole `this`? Although `waiting_for_first_frame_` is thread safe, many things under `this` are not. Having `this` captured will makes it easier to run into such thread-safety-traps in the future.

For example, someone might think: why capturing `rasterizer` instead of just using `rasterizer_` since `this` is captured? And if that person removes the `rasterizer`, we'll have a thread safety issue.

Reply:
Done.